Validate IP before resolving comment author domain

pfefferle opened the pull request:
Added a check to ensure the comment author's IP address is present and valid before attempting to resolve the hostname with gethostbyaddr. This prevents potential errors when the IP is missing or invalid.

Fixes https://wordpress.org/support/topic/php-error-warning-17/

## Proposed changes:
- Added IP validation using `filter_var()` with `FILTER_VALIDATE_IP` before calling `gethostbyaddr()`
- Initialized `$comment_author_domain` variable to an empty string as a fallback
- Wrapped the hostname resolution in a conditional check to ensure it only runs with valid IPs

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [X] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [X] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Added a safety check to prevent errors when resolving comment author hostnames without a valid IP address.

</details>

pfefferle committed:
Validate IP before resolving comment author domain

Added a check to ensure the comment author's IP address is present and valid before attempting to resolve the hostname with gethostbyaddr. This prevents potential errors when the IP is missing or invalid.

matticbot committed:
Add changelog

obenland commented on includes/class-mailer.php:
```diff
@@ -87,7 +87,11 @@ public static function comment_notification_text( $message, $comment_id ) {
 		$post                  = \get_post( $comment->comment_post_ID );
-		$comment_author_domain = \gethostbyaddr( $comment->comment_author_IP );
+		$comment_author_domain = '';
+		// Only attempt to resolve hostname if we have a valid IP address.
```
```suggestion

		// Only attempt to resolve hostname if we have a valid IP address.
```
I'm a big fan of setting defaults and conditionally changing them, and doing that in one block to make it easy to see they belong together!

Unless there's an inline comment, where making it easy to catch takes precedence for me? Let's give comments room to breath

obenland commented on includes/class-mailer.php:
```diff
@@ -87,7 +87,11 @@ public static function comment_notification_text( $message, $comment_id ) {
-		$comment_author_domain = \gethostbyaddr( $comment->comment_author_IP );
+		$comment_author_domain = '';
+		// Only attempt to resolve hostname if we have a valid IP address.
+		if ( ! empty( $comment->comment_author_IP ) && \filter_var( $comment->comment_author_IP, FILTER_VALIDATE_IP ) ) {
```
```suggestion
		if ( filter_var( $comment->comment_author_IP, FILTER_VALIDATE_IP ) ) {
```
Presumably the `filter_var()` check on its own is enough? Not sure if there's a speed concern where we'd want to avoid calling it when there's an empty value.

obenland reviewed (dismissed):
I think this is good to go as-is.

There are two optional, OCD-satisfying suggestions if you feel inclined, but I consider them optional.

pfefferle committed:
Update includes/class-mailer.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update includes/class-mailer.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Merge branch 'trunk' into fix/ip-addr

pfefferle committed:
Use global namespace for filter_var in Mailer

Prefixed filter_var with a backslash to ensure the global PHP function is used when validating IP addresses in the Mailer class.

obenland reviewed (approved):
💖
